### PR TITLE
fix: generate valid URL route paths for pages on Windows

### PR DESCRIPTION
**Pull request by petebacondarwin**
Previously route paths were manipulated by file-system path utilities.
On Windows this resulted in URLs that had backslashes, which are invalid for such URLs.

Fixes cloudflare#51
Closes cloudflare#235
Closes cloudflare#330
Closes cloudflare#327

### CODE REVIEW

**Comment by petebacondarwin** [comment]
> No changeset? Probably deserves one.

LOL! I had written one, but I didn't include it in the commit.

Fixed.

**Review by threepointone — APPROVED** [review]
The code looks good to me, stamping.

**Comment by petebacondarwin** [comment]
Thanks @threepointone - I'll let @GregBrimble take a look before landing this.

**Review by GregBrimble — APPROVED** [review]
Looks much cleaner, thanks @petebacondarwin !

**Pull request merged**
